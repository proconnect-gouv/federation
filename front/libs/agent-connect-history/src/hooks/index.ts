/* istanbul ignore file */

// declarative file
export * from './use-add-to-user-history.hook';
export * from './use-remove-from-user-history.hook';
export * from './use-user-history.hook';
