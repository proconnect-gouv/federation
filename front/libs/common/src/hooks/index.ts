/* istanbul ignore file */

// declarative file
export * from './use-api-get.effet';
