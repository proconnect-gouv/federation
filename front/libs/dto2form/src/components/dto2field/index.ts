/* istanbul ignore file */

// declarative file
export * from './dto2field.component';
