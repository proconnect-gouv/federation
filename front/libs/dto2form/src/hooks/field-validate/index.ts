export * from './field-validate.hook';
