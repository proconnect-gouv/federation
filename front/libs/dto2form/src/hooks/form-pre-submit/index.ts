export * from './form-pre-submit.hook';
