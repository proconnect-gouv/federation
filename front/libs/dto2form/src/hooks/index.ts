export * from './field-validate';
export * from './form-pre-submit';
