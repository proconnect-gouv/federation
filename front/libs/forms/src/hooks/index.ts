export * from './field-label';
export * from './field-meta';
