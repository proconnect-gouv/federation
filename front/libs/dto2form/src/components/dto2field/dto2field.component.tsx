import React from 'react';

import { Sizes } from '@fc/dsfr';
import { ChoiceField, FieldTypes, InputField, SelectField } from '@fc/forms';

import { useFieldValidate } from '../../hooks';
import type { JSONFieldType } from '../../types';

interface DTO2FieldComponentProps {
  field: JSONFieldType;
}

export const DTO2FieldComponent = React.memo(({ field }: DTO2FieldComponentProps) => {
  const { disabled, required, validators } = field;
  const validate = useFieldValidate({
    disabled,
    required,
    validators,
  });

  const config = {
    clipboardDisabled: false,
    hint: field.placeholder,
    inline: true,
    label: field.label,
    maxChars: field.maxChars,
    name: field.name,
    required: field.required,
    size: Sizes.MEDIUM,
    value: field.value,
  };

  const { type } = field;
  const choices = field.options || [];

  const isSelectField = type === FieldTypes.SELECT;
  if (isSelectField) {
    return <SelectField choices={choices} config={config} validate={validate} />;
  }

  const isChoiceField = type === FieldTypes.RADIO || type === FieldTypes.CHECKBOX;
  if (isChoiceField) {
    return <ChoiceField choices={choices} config={config} type={type} validate={validate} />;
  }

  return <InputField config={config} type={type} validate={validate} />;
});

DTO2FieldComponent.displayName = 'DTO2FieldComponent';
