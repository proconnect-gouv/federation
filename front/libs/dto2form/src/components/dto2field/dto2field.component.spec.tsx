import { render } from '@testing-library/react';

import { ChoiceField, FieldTypes, InputField, SelectField } from '@fc/forms';

import { useFieldValidate } from '../../hooks';
import { DTO2FieldComponent } from './dto2field.component';

jest.mock('../../hooks/field-validate/field-validate.hook');

describe('DTO2FieldComponent', () => {
  // given
  const maxCharsMock = Symbol('maxChars') as unknown as number;
  const disabledMock = Symbol('disabled') as unknown as boolean;
  const requiredMock = Symbol('required') as unknown as boolean;
  const placeholderMock = Symbol('placeholder') as unknown as string;
  const labelMock = Symbol('label') as unknown as string;
  const nameMock = Symbol('name') as unknown as string;
  const valueMock = Symbol('value') as unknown as string;
  const typeMock = Symbol('type') as unknown as FieldTypes;
  const fieldMock = {
    disabled: disabledMock,
    label: labelMock,
    maxChars: maxCharsMock,
    name: nameMock,
    order: 1,
    placeholder: placeholderMock,
    required: requiredMock,
    size: 'md',
    type: typeMock,
    validators: expect.anything(),
    value: valueMock,
  };
  const validateMock = jest.fn();

  beforeEach(() => {
    // given
    jest.mocked(useFieldValidate).mockReturnValue(validateMock);
  });

  it('should call useFieldValidate hook', () => {
    // when
    render(<DTO2FieldComponent field={fieldMock} />);

    // then
    expect(useFieldValidate).toHaveBeenCalledOnce();
    expect(useFieldValidate).toHaveBeenCalledWith({
      disabled: disabledMock,
      required: requiredMock,
      validators: expect.anything(),
    });
  });

  it('should match the snapshot, should create input field (default)', () => {
    // when
    const { container } = render(<DTO2FieldComponent field={fieldMock} />);

    // then
    expect(container).toMatchSnapshot();
    expect(useFieldValidate).toHaveBeenCalledOnce();
    expect(InputField).toHaveBeenCalledOnce();
    expect(InputField).toHaveBeenCalledWith(
      {
        config: {
          clipboardDisabled: false,
          hint: placeholderMock,
          inline: true,
          label: labelMock,
          maxChars: maxCharsMock,
          name: nameMock,
          required: requiredMock,
          size: 'md',
          value: valueMock,
        },
        type: typeMock,
        validate: validateMock,
      },
      {},
    );
  });

  it('should match the snapshot, should create select field', () => {
    // given
    const optionsMock = [
      {
        label: 'option 1 label mock',
        value: 'option_1_value_mock',
      },
      {
        label: 'option 2 label mock',
        value: 'option_2_value_mock',
      },
    ];
    const selectFieldMock = { ...fieldMock, options: optionsMock, type: FieldTypes.SELECT };

    // when
    render(<DTO2FieldComponent field={selectFieldMock} />);

    // then
    expect(SelectField).toHaveBeenCalledOnce();
    expect(SelectField).toHaveBeenCalledWith(
      {
        choices: optionsMock,
        config: {
          clipboardDisabled: false,
          hint: placeholderMock,
          inline: true,
          label: labelMock,
          maxChars: maxCharsMock,
          name: nameMock,
          required: requiredMock,
          size: 'md',
          value: valueMock,
        },
        validate: validateMock,
      },
      {},
    );
  });

  it('should match the snapshot, should create a choice field when type is "radio', () => {
    // given
    const optionsMock = [
      {
        label: 'option 1 label mock',
        value: 'option_1_value_mock',
      },
      {
        label: 'option 2 label mock',
        value: 'option_2_value_mock',
      },
    ];
    const radioFieldMock = { ...fieldMock, options: optionsMock, type: FieldTypes.RADIO };

    // when
    render(<DTO2FieldComponent field={radioFieldMock} />);

    // then
    expect(ChoiceField).toHaveBeenCalledOnce();
    expect(ChoiceField).toHaveBeenCalledWith(
      {
        choices: optionsMock,
        config: {
          clipboardDisabled: false,
          hint: placeholderMock,
          inline: true,
          label: labelMock,
          maxChars: maxCharsMock,
          name: nameMock,
          required: requiredMock,
          size: 'md',
          value: valueMock,
        },
        type: 'radio',
        validate: validateMock,
      },
      {},
    );
  });

  it('should match the snapshot, should create a choice field when type is "checkbox', () => {
    // given
    const optionsMock = [
      {
        label: 'option 1 label mock',
        value: 'option_1_value_mock',
      },
      {
        label: 'option 2 label mock',
        value: 'option_2_value_mock',
      },
    ];
    const checkboxFieldMock = { ...fieldMock, options: optionsMock, type: FieldTypes.CHECKBOX };

    // when
    render(<DTO2FieldComponent field={checkboxFieldMock} />);

    // then
    expect(ChoiceField).toHaveBeenCalledOnce();
    expect(ChoiceField).toHaveBeenCalledWith(
      {
        choices: optionsMock,
        config: {
          clipboardDisabled: false,
          hint: placeholderMock,
          inline: true,
          label: labelMock,
          maxChars: maxCharsMock,
          name: nameMock,
          required: requiredMock,
          size: 'md',
          value: valueMock,
        },
        type: 'checkbox',
        validate: validateMock,
      },
      {},
    );
  });
});
