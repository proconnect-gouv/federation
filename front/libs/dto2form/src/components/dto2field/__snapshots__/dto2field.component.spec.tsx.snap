// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DTO2FieldComponent should match the snapshot, should create input field (default) 1`] = `
<div>
  <div
    data-mockid="InputField"
  />
</div>
`;
